app/decorators/user_decorator.rb

```diff
@@ -43,6 +43,10 @@ def identity_not_verified?
     !user.active_profile.present?
   end
 
+  def active_identity_for(service_provider)
+    user.active_identities.find { |identity| identity.service_provider == service_provider.issuer }
```

Comment:
Presumably this kind of thing performs all the logic in ruby and rather then the database.  Can we have SQL do this for us instead somehow? (not and active model expert).